src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -303,9 +303,13 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 				newMessagesList = newMessagesList.slice(-this.messagesLimit);
 			}
 
-			this.messages = newMessagesList.sort(
-				(a, b) => timestampToNumber(b.timestamp) - timestampToNumber(a.timestamp),
-			);
+			this.messages = [
+				...new Map(
+					newMessagesList
+						.sort((a, b) => timestampToNumber(b.timestamp) - timestampToNumber(a.timestamp))
```

Comment:
why should we sort the whole list everytime we fetch a chunk of it?
it seems like there is just something wrong with it